ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/__init__.py

```diff
@@ -0,0 +1,165 @@
+        if template_path := get(self._hostvars, "switch.interface_descriptions.underlay_port_channel_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['link'] = {
+                "peer": link_peer,
+                "peer_channel_group_id": link_peer_channel_group_id,
+                "channel_description": link_channel_description,
+            }
+            return self.template_var(template_path, template_vars)
+
+        if link_channel_description is not None:
+            link_channel_description = str(link_channel_description).upper()
+            return f"{link_channel_description}_Po{link_peer_channel_group_id}"
+
+        link_peer = str(link_peer).upper()
+        return f"{link_peer}_Po{link_peer_channel_group_id}"
+
+    def mlag_ethernet_interfaces(self, mlag_interface: str) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.mlag_ethernet_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['mlag_interface'] = mlag_interface
+            return self.template_var(template_path, template_vars)
+
+        return f"MLAG_PEER_{self._mlag_peer}_{mlag_interface}"
+
+    def mlag_port_channel_interfaces(self) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.mlag_port_channel_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            return self.template_var(template_path, template_vars)
+
+        return f"MLAG_PEER_{self._mlag_peer}_Po{self._mlag_port_channel_id}"
+
+    def connected_endpoints_ethernet_interfaces(self, peer: str = None, peer_interface: str = None) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.connected_endpoints_ethernet_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['peer'] = peer
+            template_vars['peer_interface'] = peer_interface
+            return self.template_var(template_path, template_vars)
+
+        elements = [peer, peer_interface]
+        elements = [str(element) for element in elements if element is not None]
+        return '_'.join(elements)
+
+    def connected_endpoints_port_channel_interfaces(self, peer: str = None, adapter_port_channel_description: str = None) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.connected_endpoints_port_channel_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['peer'] = peer
+            template_vars['adapter_port_channel_description'] = adapter_port_channel_description
+            return self.template_var(template_path, template_vars)
+
+        elements = [peer, adapter_port_channel_description]
+        elements = [str(element) for element in elements if element is not None]
+        return '_'.join(elements)
```

Comment:
```suggestion
        elements = [peer, adapter_port_channel_description]
        return '_'.join(str(element) for element in elements if element is not None)
```

(almost made it a oneliner but that may be less readable)